File: Reverse-Prefix-of-Word.py
class Solution(object):
    def reversePrefix(self, word, ch):
        """
        :type word: str
        :type ch: str
        :rtype: str
        """
        idx = -1
        for i in range(len(word)):
            if word[i] == ch:
                idx = i
                break
        if idx == -1:
            return word
        word = list(word)
        left, right = 0, idx
        while left < right:
            word[left], word[right] = word[right], word[left]
            left += 1
            right -= 1
        return ''.join(word)

File: Reverse-Prefix-of-Word.js
/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
var reversePrefix = function(word, ch) 
{
    let idx = -1;
    for (let i = 0; i < word.length; i++) 
    {
        if (word[i] === ch) 
        {
            idx = i;
            break;
        }
    }
    if (idx === -1) 
    {
        return word;
    }
    let arr = word.split('');
    let left = 0, right = idx;
    while (left < right) 
    {
        let temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
        left++;
        right--;
    }
    return arr.join('');
};
